### PR TITLE
Fix multiple issue according @mahesh-panchal 's comment

### PR DESCRIPTION
**Pull request by jianhong**
pull from jianhong/nf-core-hicar to nf-cocre/hicar@dev

### CODE REVIEW

File: subworkflows/local/preparegenome.nf [hunk 1]
```diff
@@ -72,22 +72,24 @@ workflow PREPARE_GENOME {
...
+        //genome size remove all N then * 78% ref:https://github.com/macs3-project/MACS/issues/299
+        gs = ch_fasta.map{
+            gs = 0.0
+            it.withReader{
+                String line
+                while( line = it.readLine() ){
+                    if( !(line ==~ /^>/) ){
+                        l = line.toLowerCase()
+                        gs += l.count("a") + l.count("c") + l.count("g") + l.count("t")
+                    }
```
Comment (mahesh-panchal) [inline subworkflows/local/preparegenome.nf:88]:
```suggestion
        //genome size remove all N then * 78% ref:https://github.com/macs3-project/MACS/issues/299
        genome_size = ch_fasta.splitFasta( record: [id: true, seqString: true] )
        .map { it.seqString.replaceAll('[^acgtACGT]','').length() }
        .sum()
```
If you use `splitFasta` you also do not need to unzip the fasta first.
According to your reference `https://github.com/macs3-project/MACS/issues/299`, the multiplier you're using (0.78) is specific to that case. 
Which definition of effective genome size are you using? https://deeptools.readthedocs.io/en/latest/content/feature/effectiveGenomeSize.html

File: workflows/hicar.nf [hunk 2]
```diff
@@ -42,11 +42,14 @@ ch_circos_config         = file("$projectDir/assets/circos.conf", checkIfExists:
 
 /*
 ========================================================================================
-    JUICER_TOOLS FILE
+    TOOLS SOURCE FILE
 ========================================================================================
 */
 
-ch_juicer_tools        = params.juicer_tools_jar ? file(params.juicer_tools_jar, checkIfExists: true) : Channel.empty()
+ch_juicer_tools              = params.juicer_tools_jar ? file(params.juicer_tools_jar, checkIfExists: true) : Channel.empty()
```
Comment (mahesh-panchal) [inline workflows/hicar.nf:49]:
```suggestion
ch_juicer_tools              = file(params.juicer_tools_jar, checkIfExists: true)
```

File: modules/local/checksums.nf [hunk 3]
```diff
@@ -22,34 +22,22 @@ process CHECKSUMS {
...
-        then
-            echo "${meta.id} has checksum ${meta.md5_2}, but we got checksum \$md5!"
-            exit 128
-        fi
+    md5sum ${prefix}_1.fastq ${prefix}_2.fastq > ${prefix}.md5
+    if [ ! -z "${meta.md5_1 ?: ''}" ] && [ ! -z "${meta.md5_2 ?: ''}" ]; then
+        cat <<-END_CHECKSUM | md5sum -c
+    ${meta.md5_1}  ${prefix}_1.fastq.gz
+    ${meta.md5_2}  ${prefix}_2.fastq.gz
+    END_CHECKSUM
```
[the remark this answers is not in the document]
Comment (jianhong) [reply modules/local/checksums.nf:40]: I removed the white space to make the multiline output file format correctly.

File: subworkflows/local/preparegenome.nf (hunk 1 above)
Comment (jianhong) [reply subworkflows/local/preparegenome.nf]: I will simply using the method 1 here. Thank you for this.

File: modules/local/checksums.nf (hunk 3 above)
[the remark this answers is not in the document]
Comment (mahesh-panchal) [reply modules/local/checksums.nf:40]: Ah, sorry. My apologies. I forgot heredoc indentation only works with tab characters. The `<<-` instructs bash to remove tab characters from the start of the line in a heredoc, but not spaces.

[the remark this answers is not in the document]
Comment (mahesh-panchal) [reply modules/local/checksums.nf:40]: An alternative might then be:
```bash
printf '%s\n' \\
    "${meta.md5_1}  ${prefix}_1.fastq.gz"   \\
    "${meta.md5_2}  ${prefix}_2.fastq.gz" | \\
    md5sum -c
```
but you choose.

**Review by mahesh-panchal — APPROVED** [review]

File: modules/local/checksums.nf (hunk 3 above)
[the remark this answers is not in the document]
Comment (jianhong) [reply modules/local/checksums.nf:40]: Thank you @mahesh-panchal . I prefer the first one.

**Pull request merged**